core/ops.rs

```diff
@@ -218,16 +219,19 @@ where
   F: Fn(Rc<RefCell<OpState>>, Value, BufVec) -> R + 'static,
   R: Future<Output = Result<Value, AnyError>> + 'static,
 {
+  #[derive(Deserialize)]
+  #[serde(rename_all = "camelCase")]
+  struct Params {
+    promise_id: u64,
+    args: Value,
+  }
```

Comment:
IIRC this was removed by @piscisaureus for the performance reasons

Reply:
Can you elaborate? We could change this to a JSON array. I think that would have less overhead than the previous impl.

Reply:
The idea was that getting only `promise_id` was faster than trying to do deserialization into struct here and then once again in the actual op. Maybe we should guard arguments passed to `jsonOpAsync` (and sync as well) in `core.js`?

Reply:
If that is the case, array should be just as fast. Let me implement that.